packages/block-editor/src/components/use-editor-fonts-resolver/index.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useMemo, useCallback } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { getDisplaySrcFromFontFace, loadFontFaceInBrowser } from './utils';
+import { store as editorStore } from '../../store';
+
+function useEditorFontsResolver() {
+	const [ loadedFontUrls, setLoadedFontUrls ] = useState( new Set() );
+
+	const { currentTheme = {}, fontFamilies = [] } = useSelect( ( select ) => {
+		return {
+			currentTheme:
+				select( editorStore ).getSettings()?.__experimentalFeatures
+					?.currentTheme,
+			fontFamilies:
+				select( editorStore ).getSettings()?.__experimentalFeatures
+					?.typography?.fontFamilies,
+		};
```

Comment:
Opportunity to call `getSettings()` once, e.g., 

```js
		const { __experimentalFeatures } =
			select( blockEditorStore ).getSettings();
		return {
			currentTheme: __experimentalFeatures?.currentTheme,
			fontFamilies: __experimentalFeatures?.typography?.fontFamilies,
		};
```

Reply:
This part of the code was removed. Marking this comment as resolved